src/event-log/event-emitter-stream.ts

```diff
@@ -3,24 +3,42 @@ import type { KeyValues } from '../types/query-types.js';
 import type { EventListener, EventStream, EventSubscription } from '../types/subscriptions.js';
 
 import { EventEmitter } from 'events';
+import { DwnError, DwnErrorCode } from '../core/dwn-error.js';
 
 const EVENTS_LISTENER_CHANNEL = 'events';
 
+export interface EventEmitterStreamConfig {
+  /**
+   * An optional error handler in order to be able to react to any errors or warnings triggers by `EventEmitter`.
+   * By default we log errors with `console.error`.
+   */
+  errorHandler?: (error: any) => void;
+};
+
 export class EventEmitterStream implements EventStream {
   private eventEmitter: EventEmitter;
   private isOpen: boolean = false;
 
-  constructor() {
+  constructor(config?: EventEmitterStreamConfig) {
```

Comment:
Optional: I kinda what you did as a pattern that I saw somewhere.

```suggestion
  constructor(config: EventEmitterStreamConfig = {}) {
```